x/examples/soax-relay/README.md

```diff
@@ -0,0 +1,131 @@
+# SOCKS5 Proxy Relay for SOAX
+
+This program creates a SOCKS5 proxy relay that allows users to share access to SOAX without sharing their own credentials. It's designed for connectivity testing purposes.
+
+## Design
+
+The proxy relay works as follows:
+
+1. It listens for incoming SOCKS5 connections on a specified local address and port.
+2. It authenticates incoming connections using a custom authentication mechanism.
+3. For authenticated connections, it relays requests to the SOAX proxy server.
+4. It supports both TCP and UDP traffic.
+
+#### Key components:
+
+- **CustomAuthenticator**: Handles authentication based on username prefix and extracts the suffix which includes connection parameters such as country, city, ISP, etc.
+- **StaticCredentialStore**: Stores and validates user credentials.
+- **UDP Associate Handler**: Manages UDP connections by relaying the SOAX bind address back to the client.
+
+## Usage
+
+### Configuration
+
+The program uses a YAML configuration file (`config.yaml`) for settings. Example configuration:
+
+```yaml
+server:
+  address: "127.0.0.1"
+  port: "1080"
+
+upstream:
+  prefix: "package-xxxxx-"
+  password: "YourSOAXPassword"
+  address: "proxy.soax.com:5000"
+
+credentials:
+  foo: "bar"
+  jane: "password123"
+
+udp_timeout: 1m
+```
+
+### Authentication
+Users connect to the relay using the following format:
+
+<b>Username:</b> 
+
+username-country-{cc_code}-sessionid-{session_id}-sessionlength-{time_in_sec}
+
+where any part that goes after username-country-{cc_code} is optional. For more information on the format refer to [this document](https://helpcenter.soax.com/en/articles/6723733-sticky-sessions) on soax support page.
+
+<b>Password:</b> (as defined in the credentials section of the config file)
+
+<b>Example Username:</b>
+
+outline-country-ru-sessionid-12-sessionlength-600
+
+The relay will use the SOAX package ID and password to connect to the SOAX server, appending the country, sessionid, and sessionlength from the user's input.
+
+### Setup and Running
```

Comment:
This is too complicated. You can just use `go run` directly and pass the full package name. See the docs for the other tools.

Reply:
I simplified the readme accordingly